### ISSUE TITLE
Notification system

### ISSUE THREAD

**Issue by omarsotillo** [TheRealWolkenbruch/concept#6]
## Background 🔙

We came into the discussion during today meeting about how we will set notifications. I think is a big topic and probably we can recap ideas of how to approach that issue 👯‍♂️

## Proposal 🚀

Using the gem [Whenever](https://github.com/javan/whenever) This gem allows to create cronjobs and specify the command job run during that time. 

For that I propose the following:
- Users are added into specific notification ranges. E.g
  - I want to be notified every day once at midday, 
  - I want to be notified every day during the night. 
  - Two days per day.
- These users are then persisted into a database table where they added a specific notification setting. Enum table for example :every_day_twice, :every_day_once, :every_two_days

- We define the following cronjobs (whenever gem example):
```
every 1.day, at: ['4:30 am', '6:00 pm'] do
  rake "notify:users:for_this_setting"
end

every :day do 
  rake "notify:users:for_this_setting"
end

every :sunday, at: '12pm' do 
  rake "notify:users:for_this_setting"
end

every '0 0 27-31 * *' do
  rake "notify:users:for_this_setting"
end
```

There are as many `every` as configurations we have enabled for the unprotected users.

This was a simple approach I came up with which could allow us to have asynchronous stuff. 

What do you think? 🤔

**Comment by hendrikb** [comment]
We have been using the [whenever gem](https://github.com/javan/whenever) for years at Aroundhome.

The thing is, implements it's scheduling logic directly into the underlying system's `crontab`. This is of course pretty evasive and mingles with the stuff that's already there. We at Aroundhome moved away from this approach after we prepared for Docker as these two are just [not _very_ compatible with each other](https://blog.kurttomlinson.com/posts/how-to-run-cron-jobs-with-the-whenever-gem-in-a-docker-container) - scheduling is now done through cloud-based services.

Long story short: This _strongly_ depends on how we plan to migrate this thing. I'm gonna wrap my head around this topic, I think `whenever` is not the perfect solution here.
